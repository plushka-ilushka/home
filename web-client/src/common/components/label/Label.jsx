import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import './label.scss';

export default function Label(props) {
  const {
    className,
    children,
    name,
    caption,
    captionClassName,
  } = props;

  return (
    <label
      className={classNames('label', className)}
      htmlFor={name}
    >
      <div className={captionClassName}>{caption}</div>
      {children}
    </label>
  );
}

Label.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  name: PropTypes.string,
  caption: PropTypes.string,
  captionClassName: PropTypes.string,
};

Label.defaultProps = {
  className: undefined,
  name: undefined,
  caption: undefined,
  captionClassName: undefined,
};
