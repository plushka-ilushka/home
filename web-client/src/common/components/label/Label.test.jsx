import React from 'react';
import { mount } from 'enzyme';
import Label from './Label';

describe('Label Component', () => {
  it('Label Snaphot', () => {
    expect(mount(<Label className="a">children</Label>)).toMatchSnapshot();
  });
});
