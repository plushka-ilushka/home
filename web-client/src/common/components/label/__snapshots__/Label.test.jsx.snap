// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component Label Snaphot 1`] = `
<Label
  className="a"
>
  <label
    className="label a"
  >
    children
  </label>
</Label>
`;
